src/d3d9/d3d9_state.h

```diff
@@ -245,19 +245,23 @@ namespace dxvk {
           bool                 FloatEmu) {
     auto UpdateHelper = [&] (auto& set) {
       if constexpr (ConstantType == D3D9ConstantType::Float) {
-        auto begin = reinterpret_cast<const Vector4*>(pConstantData);
-        auto end   = begin + Count;
 
-        if (!FloatEmu)
-          std::copy(begin, end, &set.fConsts[StartRegister]);
-        else
-          std::transform(begin, end, &set.fConsts[StartRegister], replaceNaN);
+        if (!FloatEmu) {
+          size_t len = Count * sizeof(Vector4);
+          void* dest = reinterpret_cast<void*>(&set.fConsts[StartRegister]);
```

Comment:
`reinterpret_cast<void*>` makes no sense

Reply:
it's there to avoid a warning
```
warning: 'void* memcpy(void*, const void*, size_t)' copying an object of non-trivial type 'using Vector4i = struct dxvk::Vector4Base<int>' {aka 'struct dxvk::Vector4Base<int>'} from an array of 'const int' [-Wclass-memaccess]
  251 |         std::memcpy(&set.fConsts[StartRegister], pConstantData, len);
```
open for suggestions what do instead.

Reply:
ah I guess I can just use 
```c
fConsts[StartRegister].data
```
:frog: